### PR TITLE
Add FAQ section to docs

### PR DESCRIPTION
**Pull request by sgarcez**
As suggested by @jfuss here is an initial stab at an FAQ section for the docs.

Currently contains only a couple of questions - how to deal with multiple environments/stages, and how to enable logs in gateway.

Let me know if this useful in its present form or whether the official recommendations are different from what I thought.

### CODE REVIEW

**Comment by jfuss** [comment]
This is awesome!!!! I will take a look later today.

File: docs/faq.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+This approach limits the 'blast radius' for any given deployment since all resources for each environment are scoped to a different CloudFormation Stack you will never be editing production resources on accident.
+
+If you need to bring up separate stacks for different reasons (multiple region deployments, developer/branch stacks) it will be straightforward to do so with this approach since the same template can be used to bring up and manage a new stack independent of any others.
+
+In cases where you need to manage different stages differently this can be done through a combination of Stack Parameters, Conditions, and Fn::If statements.
+
+How to enable API Gateway Logs
+------------------------------
+
+Work is underway to make this functionality part of the SAM specification. Until then the suggested workaround is to use the ``aws cli update-stage`` command to enable this manually after a Clouformation deployment.
```
Comment (jfuss) [inline docs/faq.rst]: Can you add a note about the following:

If you are to follow this, you will see metric gaps between the time CloudFormation deployed and the time you run this command. 

Might be worth suggesting another option would be to create a custom resource that does this as well. I do know of customers going this route.

File: docs/cloudformation_compatibility.rst [hunk 2]
```diff
@@ -1,5 +1,5 @@
 CloudFormation Compatibility Section
-===============
+====================================
```
[the remark this answers is not in the document]
Comment (sgarcez) [reply docs/cloudformation_compatibility.rst]: Just the warnings from `sphinx-build` when running `make html` :)

[the remark this answers is not in the document]
Comment (jfuss) [reply docs/cloudformation_compatibility.rst]: Nice. Will keep that in mind in the future :)

**Comment by jfuss** [comment]
Thank you for doing this. Great addition :)

Sorry it took me some time. I was out of the office for part of last week.

**Pull request merged**